contrib/nixos/default.nix

```diff
@@ -20,6 +25,51 @@ let
     map (x: "${x.from} ${x.to}") cfg.mimeMap);
 
   cfg = config.services.gosh;
+
+  goshConfig = pkgs.writeText "gosh.yml" ''
```

Comment:
Please take a look at `pkgs.formats.toml`. One could drastically simplify the module by just passing the configuration as a `types.attrs` or `types.submodule`.